File: array/java/twoSum.java
/*
Example:
Given nums = [2, 7, 11, 15], target = 9,

Because nums[0] + nums[1] = 2 + 7 = 9,
return [0, 1].
*/


public class TwoSum
{
    public static int[] twosum(int[] nums, int target)
    {
         int res[] = new int[2]; 
         for (int i=0; i<nums.length; i++)
         {
              for(int j=i+1; j<nums.length; j++)
              {
                   if (nums[i] + nums[j] == target)
                   {
                       /* res[0] = i;
                       res[1] = j;
                       return res;
                       */
                       return new int[] {i, j};
                   }

              }
         }
         // return res;
         throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String []args)
    {
        int nums[] = {2, 7, 11, 15};
        int target = 9;
        twosum(nums, target);
    }
}

File: array/python/twosum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in range(len(nums)):
            for j in range(1, len(nums)):
                if target == nums[i] + nums[j]:
                    list = [i, j]
                    return list


if __name__ == "__main__":
    solution = Solution()
    list = solution.twoSum([1, 2, 7, 5], 9)
    print list 
    print list[0]
    print list[1]